### DISCUSSION TITLE
is git LFS compatible with automatic pipelines?

### DISCUSSION THREAD

**Discussion by ggrrll** [git-lfs/git-lfs#5458] [Q&A]
Hi, 

if I understood, LFS replaces files with pointers, so this makes it impossible for an automatic pipeline to work with LFS-tracked files. 
is this correct? or am I missing something?   is there a workaround ? 

thanks

**Comment by chrisd8088** [comment]
Can you provide a specific example?  I'm not too sure what you mean by "automatic pipeline".  Lots of projects, including ours, have CI jobs which use Git LFS successfully, for instance.

**Comment by ggrrll** [comment]
I mean, if you have scripts that use as input LFS-tracked files, then they won't work anymore, as those inputs are reduced to pointers... (right?)

**Comment by chrisd8088** [comment]
If you have scripts which interact with Git, it depends on how they work -- if Git LFS is installed and configured, then it Git will run it automatically when you check out objects into a working tree, and any Git LFS pointers will be replaced with the full file contents.

It really depends on what you're doing and how you're interacting with the repository contents.

**Reply by ggrrll** [reply]
yeah, but the checkout has be done manually, or inserted in the pipeline explicitly  for those files needed , right? ( `migrate export...` )

so, one has so continuously re-`export` and re-`import` files as they are needed (correct?) -- if you still want to push those modified files on LFS

**Reply by ggrrll** [reply]
ok, **something weird** happened while testing:

yesterday I installed  LFS and tracked files on a repo (on an external drive), without committing  and _files were replaced with pointers_

today, I did the same on another repo, following https://git-lfs.com/, but files were _not_ replaced ; instead they were correctly replaced on another clone , on a different machine

I guess the correct behaviour is the 2nd one ... should I open an issue about the first one?

**Comment by chrisd8088** [comment]
> yeah, but the checkout has be done manually, or inserted in the pipeline explicitly for those files needed , right?

I don't think so ... you are welcome to browse the many testing scripts in https://github.com/git-lfs/git-lfs/tree/main/t which use Git and Git LFS extensively, and generally without needing to import or export.  Those commands are only needed if you are taking an existing repository and Git history without Git LFS and converting it to use Git LFS, or vice versa.

**Comment by chrisd8088** [comment]
> yesterday I installed LFS and tracked files on a repo (on an external drive), without committing and _files were replaced_ with pointers

Can you provide a complete reproduction case?

**Reply by ggrrll** [reply]
#5463
